### PR TITLE
Github verification

### LINKED ISSUE #31: Prevent building and publishing plugins from different repositories other than authors
Right now, there's nothing preventing someone from linking repositories under different GitHub accounts and then publishing plugins. This means there could be multiple "by Kukks" plugins published by different Plugin Builder accounts - some of which Kukks doesn't control.

### PR DESCRIPTION
**Pull request by TChukwuleta**
Resolves #31

### CODE REVIEW

File: PluginBuilder/Views/Account/GithubAccountVerification.cshtml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        <div class="alert alert-info">
+            <p>
+                To verify your GitHub account, please create a public
+                <a href="https://gist.github.com/" target="_blank" style="color: inherit; text-decoration: underline;">gist</a>
+                with the following token:
+            </p>
+            <h5><span id="token">@Model.Token</span></h5>
+            <p>Copy the token above and create a new public gist on your GitHub account with this token as the content.</p>
+            <p>Once done, submit the Gist ID for validation.</p>
+        </div>
```
Comment (rockstardev) [inline PluginBuilder/Views/Account/GithubAccountVerification.cshtml:21]:
```suggestion
        <div class="alert alert-info">
        To verify your GitHub account, <a href="https://gist.github.com/" target="_blank" style="color: inherit; text-decoration: underline;">create a public gist</a> with the following content:<br>
        <span id="token" class="fs-3">ca9b473b-5e3a-480d-886e-0cb7136e81d3</span><br><br>
        The gist description and filename do not matter, just ensure the provided token is entered as the content.<br>
        Once the public gist is created, copy its full URL and click the "Verify" button to complete verification.
        </div>
```

Proposed improved text

**Review by rockstardev — CHANGES_REQUESTED** [review]
We need the GitHub verification flow to mirror the Email verification flow as demonstrated here:

[[VIDEO URL]](https://github.com/user-attachments/assets/ed9f3e31-2407-43e8-8f1b-112361f669b1)

The main difference will be that the `Verify` button for GitHub will redirect users to the new page you have built. After successful verification, it should redirect back to the Account Details page, displaying a `Confirmed` label next to GitHub.

---

**Changes Needed for Verifying GitHub Account:**

Here's the current setup:

image

1. Combine the Github Profile Url and Public Gist Id into a single textbox - Public Gist Url. From this URL, we will extract the GitHub username.
2. The form needs better formatting—make the width similar to the Account Details form.
3. The instructions should be clearer. I will provide a proposed commit with the revised text.

---

**Database Changes for GitHub Verification:**

When saving GitHub verification in the database, instead of a simple boolean, save the full public URL of the Gist. This allows us to periodically check if the Gist still exists—if the user deletes the Gist, we will revoke verification.

---

**Remove Separate Trigger for GitHub Account Verification:**

Finally, eliminate the separate blocker/trigger for GitHub account verification during the build. The trigger will now be in the Account Details, so there’s no need for this check to be added here. For now, we want users to continue being able to publish new versions without verifying, and we can implement these checks later.

**Comment by rockstardev** [comment]
Also in future please avoid adding multiple reviewers to prevent unnecessary work. For this repository, request a review from me... and once we reach the appropriate milestones, we can present it to the rest of the team together.

**Comment by TChukwuleta** [comment]
@rockstardev  flow has been updated

**Review by rockstardev — APPROVED** [review]
LGTM and thanks!

**Pull request merged**